src/extensions/ThumbnailsExtension/ThumbnailsExtension.ts

```diff
@@ -43,6 +42,43 @@ export class ThumbnailsExtension extends Module {
       .sort((a, b) => b.start - a.start);
   }
 
+  private async loadBifThumbs(file) {
+    const response = await fetch(file);
+    const data = await response.arrayBuffer();
+
+    const BifParser = new BIFParserClass(data);
```

Comment:
Might be worth reading up on how we can write our own BIF parser (seems fairly easy to do, https://developer.roku.com/en-gb/docs/developer-program/media-playback/trick-mode.md).

Wouldn't it be possible to import the parser directly from videojs-bif? This way, we preserve the source of the parser.

Edit: Noticed that they have the same API as ours with `getThumbnail(seconds): IThumbnail` here: https://github.com/chemoish/videojs-bif/blob/master/src/parser.js#L125 This way, we don't keep a reference in memory for each thumbnail upfront.

Reply:
I set up the API for getting images by the second [here](https://github.com/matvp91/indigo-player/compare/master...stoicbuddha:bifupdates?expand=1#diff-141cf3ef42476a2635b04e8fa2d206f2R77). Note that I updated [the method](https://github.com/matvp91/indigo-player/compare/master...stoicbuddha:bifupdates?expand=1#diff-a7e5392951c4211ea2d653559292f8d9R150) for getting BIF image data at a given second because the original method didn't work at all.